staging/src/k8s.io/cli-runtime/pkg/genericclioptions/resource/visitor.go

```diff
@@ -463,7 +471,10 @@ func ExpandPathsToFileVisitors(mapper *mapper, paths string, recursive bool, ext
 		if path != paths && ignoreFile(path, extensions) {
 			return nil
 		}
-
+		if filepath.Base(path) == constants.KustomizationFileName {
```

Comment:
This will be changed to check the GVK of the kustomization file in an immediate follow up.